src/kokkos_abstraction.hpp

```diff
@@ -52,6 +53,26 @@ using ParArray5D = Kokkos::View<T *****, LayoutWrapper, DevMemSpace>;
 template <typename T>
 using ParArray6D = Kokkos::View<T ******, LayoutWrapper, DevMemSpace>;
 
+template <bool>
+struct UseLightweightKernel {};
+
+template <typename PolicyType>
+auto par_policy(PolicyType const &policy_old, UseLightweightKernel<true>) {
```

Comment:
Food for thought along these lines - maybe we should get rid of all the explicit references to `MDRangePolicy` and instead have the instantiate a `parthenon::md_range_policy` that we can control at compile time to switch between normal `RangePolicy` and `MDRangePolicy`, similar to `par_for`. Doesn't need to be done here, but something to consider.

Reply:
Yes, I agree. I had a similar thought and thing that it'd make sense (as a separate issue).